File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOScientificSleepSwitchParamModel.swift
//
//  IDOScientificSleepSwitchParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Scientific sleep switch setting event
@objcMembers
public class IDOScientificSleepSwitchParamModel: NSObject, IDOBaseModel {
    /// Mode
    /// 2: Scientific sleep
    /// 1: Normal sleep
    public var mode: Int
    /// Start time - hour
    public var startHour: Int
    /// Start time - minute
    public var startMinute: Int
    /// End time - hour                                    
    public var endHour: Int
    /// End time - minute
    public var endMinute: Int
    
    enum CodingKeys: String, CodingKey {
        case mode = "mode"
        case startHour = "start_hour"
        case startMinute = "start_minute"
        case endHour = "end_hour"
        case endMinute = "end_minute"
    }
    
    public init(mode: Int,startHour: Int,startMinute: Int,endHour: Int,endMinute: Int) {
        self.mode = mode
        self.startHour = startHour
        self.startMinute = startMinute
        self.endHour = endHour
        self.endMinute = endMinute
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOScientificSleepSwitchParamModel.kt
package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

class IDOScientificSleepSwitchParamModel(
    mode: Int,
    startHour: Int,
    startMinute: Int,
    endHour: Int,
    endMinute: Int
) : IDOBaseModel {

    /// Mode
    /// 2: Scientific sleep
    /// 1: Normal sleep
    @SerializedName("mode")
    var mode: Int = mode

    /// Start time - hour
    @SerializedName("start_hour")
    var startHour: Int = startHour

    /// Start time - minute
    @SerializedName("start_minute")
    var startMinute: Int = startMinute

    /// End time - hour
    @SerializedName("end_hour")
    var endHour: Int = endHour

    /// End time - minute
    @SerializedName("end_minute")
    var endMinute: Int = endMinute

    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }

}